src/engine/GraphStoreProtocol.cpp

```diff
@@ -46,35 +46,51 @@ std::vector<TurtleTriple> GraphStoreProtocol::parseTriples(
 }
 
 // ____________________________________________________________________________
-std::vector<SparqlTripleSimpleWithGraph> GraphStoreProtocol::convertTriples(
-    const GraphOrDefault& graph, std::vector<TurtleTriple> triples) {
+updateClause::GraphUpdate::Triples GraphStoreProtocol::convertTriples(
+    const GraphOrDefault& graph, std::vector<TurtleTriple> triples,
+    Quads::BlankNodeAdder& blankNodeAdder) {
   SparqlTripleSimpleWithGraph::Graph tripleGraph{std::monostate{}};
   if (std::holds_alternative<GraphRef>(graph)) {
     tripleGraph = std::get<GraphRef>(graph);
   }
-  auto transformTurtleTriple = [&tripleGraph](TurtleTriple&& triple) {
+  auto transformTc =
+      [&blankNodeAdder](const TripleComponent& tc) -> TripleComponent {
+    if (tc.isString()) {
+      AD_CORRECTNESS_CHECK(tc.getString().starts_with("_"));
+      // Remove the `_:`.
+      return blankNodeAdder.getBlankNodeIndex(tc.toString().substr(2));
```

Comment:
Can you refactor the remaining other usage of the blank node adder, s.t. the blank node
adder requires the input to start with `_:` and removes it iself (reduce code duplication)?
Also there is a copy+paste bug (starts_with("_:")).